src/Entity/Member.php

```diff
@@ -0,0 +1,165 @@
+<?php
+
+namespace App\Entity;
+
+use ApiPlatform\Doctrine\Orm\Filter\SearchFilter;
+use ApiPlatform\Metadata\ApiFilter;
+use ApiPlatform\Metadata\ApiResource;
+use ApiPlatform\Metadata\Delete;
+use ApiPlatform\Metadata\Get;
+use ApiPlatform\Metadata\GetCollection;
+use ApiPlatform\Metadata\Post;
+use ApiPlatform\Metadata\Put;
+use App\Repository\MemberRepository;
+use App\Validator\Constraints\EventMember;
+use App\Validator\Constraints\MemberAndPayerSameEvent;
+use App\Validator\Constraints\MemberIsNotPayer;
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Doctrine\ORM\Mapping as ORM;
+use Symfony\Component\Serializer\Annotation\Groups;
+use Symfony\Component\Validator\Constraints as Assert;
+
+#[ApiResource(
+    operations: [
+        new Get(),
+        new GetCollection(),
+        new Post(
+            denormalizationContext: ['groups' => 'member:create'],
+            validationContext: ['groups' => 'member:create']
+        ),
+        new Put(
+            denormalizationContext: ['groups' => 'member:update'],
+        ),
+        new Delete(),
+    ],
+    formats: ['json'],
+    normalizationContext: ['groups' => 'member:read'],
+    denormalizationContext: [
+        'groups' => [
+            'member:create',
+            'member:update'
+        ]
+    ],
+    collectDenormalizationErrors: true
+)]
+#[ApiFilter(filterClass: SearchFilter::class, properties: ['user', 'event', 'payer'])]
+#[ORM\Entity(repositoryClass: MemberRepository::class)]
+#[ORM\UniqueConstraint(columns: ['user_uid', 'event_uid'])]
+class Member
+{
+    #[ORM\Id]
+    #[ORM\Column(length: 255, unique: true)]
+    #[Assert\NotBlank]
+    #[Groups(['member:create', 'member:read'])]
+    private string $uid;
+
+    #[ORM\ManyToOne]
+    #[ORM\JoinColumn(name: 'user_uid', referencedColumnName: 'uid', nullable: false, onDelete: "CASCADE")]
```

Comment:
### ❗ Issue: Don't delete it. Set null instead

Even if someone delete user - we should keep all event invormation saved. Thats why this field can be null :)

Reply:
need to add validation, and avoid deleting the member with assigned payments